File: Spring Week 2/templates/A/A_ants.java
import java.io.*;
import java.util.*;

public class Main{
    // Parameters:
    // Input N, L as described in the problem statement
    // positions, an array of length N containing the positions of the ants
    // Output:
    // Array of two integers as described in the problem statement
    public static int[] ants(int N, int L, int[] positions){
        int[] result = new int[2];
        // Your code here
        return result;
    }

    public static void main(String[] args){
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int Q = Integer.parseInt(br.readLine());
        StringTokenizer st = new StringTokenizer(br.readLine());
        for(int i = 0; i < Q; i++){
            int L = Integer.parseInt(st.nextToken());
            int N = Integer.parseInt(st.nextToken());
            int[] positions = new int[N];
            for(int j = 0; j < N; j++){
                positions[j] = Integer.parseInt(st.nextToken());
            }
            int[] result = ants(N, L, positions);
            System.out.println(result[0] + " " + result[1]);
        }
    }
    static int parseInt(StringTokenizer st, BufferedReader br) throws IOException{
        while(!st.hasMoreTokens()){
            st = new StringTokenizer(br.readLine());
        }
        return Integer.parseInt(st.nextToken());
    }
}

File: Spring Week 2/templates/A/A_ants.cpp
#include <bits/stdc++.h>
using namespace std;

// Parameters:
// Input N,L as described in the problem statement
// positions: vector of length N containing the positions of the ants
// Output: pair<int,int> as described in the problem statement
pair<int,int> ants(int N, int L, vector<int> positions) {
    pair<int,int> ans;

    return ans;
}
int main(){
    int Q;
    cin >> Q;
    while(Q-- > 0){
        int N,L;    
        cin >> L >> N;
        vector<int> positions(N);
        for(int i = 0; i < N; i++){
            cin >> positions[i];
        }
        pair<int,int> ans = ants(N,L,positions);
        cout << ans.first << " " << ans.second << endl;
    }
}